### PR TITLE
Add support for Secrets Manager

### PR DESCRIPTION
**Pull request by duncanista**
### What does this PR do?

Add support for Secrets Manager to the datadog serverless plugin.

### Motivation

[SLS-1553](https://datadoghq.atlassian.net/browse/SLS-1553)

### Testing Guidelines

Added a line to test which checks all the environment variables.

### Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,15 +27,16 @@ To further configure your plugin, use the following custom parameters in your `s
...
-| `flushMetricsToLogs`        | Send custom metrics by using logs with the Datadog Forwarder Lambda function (recommended). Defaults to `true`. If you disable this parameter, it's required to set `apiKey` (or `apiKMSKey` if encrypted). `flushMetricsToLogs` is ignored when `addExtension` is true.                                                                                                                                                                 |
+| `flushMetricsToLogs`        | Send custom metrics by using logs with the Datadog Forwarder Lambda function (recommended). Defaults to `true`. If you disable this parameter, it's required to set `apiKey` (or `apiKMSKey` if encrypted, or `apiKeySecretArn`). `flushMetricsToLogs` is ignored when `addExtension` is true.                                                                                                                                           |
```
Comment (ruthnaebeck) [inline README.md:30]:
```suggestion
| `flushMetricsToLogs`        | Send custom metrics by using logs with the Datadog Forwarder Lambda function (recommended). Defaults to `true`. If you disable this parameter, it's required to set `apiKey` or if encrypted, `apiKMSKey` or `apiKeySecretArn`. `flushMetricsToLogs` is ignored when `addExtension` is true.                                                                                                                                           |
```

File: README.md [hunk 2]
```diff
@@ -27,15 +27,16 @@ To further configure your plugin, use the following custom parameters in your `s
...
-| `addExtension`              | Whether to install the Datadog Lambda Extension as a layer. Defaults to `false`. When enabled, it's required to set the `apiKey` (or `apiKMSKey`) parameter. Learn more about the Lambda Extension Layer [here][8]. Note: AWS only supports Lambda Extensions for [certain runtimes][9].                                                                                                                                                 |
+| `addExtension`              | Whether to install the Datadog Lambda Extension as a layer. Defaults to `false`. When enabled, it's required to set the `apiKey` (or `apiKMSKey` or `apiKeySecretArn`) parameter. Learn more about the Lambda Extension Layer [here][8]. Note: AWS only supports Lambda Extensions for [certain runtimes][9].                                                                                                                            |
```
Comment (ruthnaebeck) [inline README.md:39]:
```suggestion
| `addExtension`              | Whether to install the Datadog Lambda Extension as a layer. Defaults to `false`. When enabled, it's required to set the `apiKey`, `apiKMSKey`, or `apiKeySecretArn` parameter. Learn more about the [Datadog Lambda Extension Layer][8] in the documentation. **Note**: AWS only supports Lambda Extensions for [certain runtimes][9].                                                                                                                            |
```

File: src/env.ts [hunk 3]
```diff
@@ -105,6 +108,10 @@ export function setEnvConfiguration(config: Configuration, handlers: FunctionInf
     if (config.apiKMSKey !== undefined && environment[apiKeyKMSEnvVar] === undefined) {
       environment[apiKeyKMSEnvVar] = config.apiKMSKey;
     }
+    const isNode = runtimeLookup[handler.runtime!] === RuntimeType.NODE;
+    if (config.apiKeySecretArn !== undefined && environment[apiKeySecretArnEnvVar] === undefined && !isNode) {
```
[the remark this answers is not in the document]
Comment (duncanista) [reply src/env.ts:112]: Oh, it's not only for Node functions, we just don't add the `DD_API_KEY_SECRET_ARN` when the function is not Node.

[the remark this answers is not in the document]
Comment (duncanista) [reply src/env.ts]: Resolved with the details of our yesterdays' chat

File: README.md [hunk 4]
```diff
@@ -33,15 +33,16 @@ To further configure your plugin, use the following custom parameters in your `s
...
+| `flushMetricsToLogs`        | Send custom metrics by using logs with the Datadog Forwarder Lambda function (recommended). Defaults to `true`. If you disable this parameter, it's required to set `apiKey` or if encrypted, `apiKMSKey` or `apiKeySecretArn`. `flushMetricsToLogs` is ignored when `addExtension` is true.                                                                                                                                             |
+| `site`                      | Set which Datadog site to send data, this is only used when `flushMetricsToLogs` is `false` or `addExtension` is `true`. Possible values are `datadoghq.com`, `datadoghq.eu`, `us3.datadoghq.com`, `us5.datadoghq.com`, and `ddog-gov.com`. The default is `datadoghq.com`.com`.                                                                                                                                                         |
```
Comment (nhinsch) [inline README.md:37]: Looks like there is a typo at the end of this line

File: src/index.ts [hunk 5]
```diff
@@ -250,26 +250,23 @@ function configHasOldProperties(obj: any) {
 }
 
 function validateConfiguration(config: Configuration) {
+  checkForMultipleApiKeys(config);
```
Comment (nhinsch) [inline src/index.ts:253]: 👌

**Review by nhinsch — APPROVED** [review]
Looks great!

File: README.md (hunk 4 above)
Comment (duncanista) [reply README.md:37]: True, missed it, thank you!

**Pull request merged**